metrics/metrics.go

```diff
@@ -202,9 +201,8 @@ func InitMetrics(
 		failOnError     = false // whether the agent will fail to start if an error happened during init metric source
 
 		metricsSources = []MetricsSource{}
-		foundErrors    = make([]error, 0)
 	)
-
+	var foundErrors error
```

Comment:
This would be more readable if it's kept as a list and the list converted to string in the end.